### PR TITLE
Add file to include the transpiled check

### PR DESCRIPTION
**Pull request by DylanC**
Moving checkForTranspiledCode() to a common location for access from all the individual tools.

### CODE REVIEW

**Comment by DylanC** [comment]
Yeah, exporting would be smart! 😆 I'll update the readme and add a test too.

File: lib/source-check.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  // Check for a source map
+  isTranspiled = readFile.includes('//# sourceMappingURL=')
+
+  while ((matchedObj = regex.exec(readFile)) !== null) {
+    // Avoid infinite loops with zero-width matches
+    if (matchedObj.index === regex.lastIndex) {
+      regex.lastIndex++
+    }
+    // Loop through results and check length of fn name
+    matchedObj.forEach((match, groupIndex) => {
```
Comment (goto-bus-stop) [inline lib/source-check.js]: Do we need to loop here? I think matchedObj is an array-like thing containing `[fullMatch, firstGroup]`, so we could only check for `matchedObj[1]`.

File: lib/source-check.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      if (groupIndex !== 0 && match.length < 3) {
+        isTranspiled = true
```
Comment (goto-bus-stop) [inline lib/source-check.js]: Considering it transpiled if just one function name is short is maybe a bit too strict. For example, the `assert` module contains an `ok` function, but it's not transpiled.

File: lib/source-check.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+function checkForTranspiledCode (filename) {
+  const readFile = fs.readFileSync(filename, 'utf8')
+  const regex = /function\s+(\w+)/g
+  let matchedObj
+  let isTranspiled = false
+
+  // Check for a source map
+  isTranspiled = readFile.includes('//# sourceMappingURL=')
+
+  while ((matchedObj = regex.exec(readFile)) !== null) {
```
Comment (goto-bus-stop) [inline lib/source-check.js]: We should maybe only do this if the source map comment was not found. There's no possibility that the loop decides that it's actually _not_ transpiled.

File: lib/source-check.js (hunk 1 above)
Comment (DylanC) [reply lib/source-check.js]: Ah you are right. I've removed it since it was overly complicated.

File: lib/source-check.js (hunk 3 above)
Comment (DylanC) [reply lib/source-check.js]: By default isTranspiled is false.  `let isTranspiled = false`. Either the source check or map comment check will set it to true.

I could change it to :
`If(readFile.includes('//# sourceMappingURL=')) { isTranspiled = true }`

Better?

File: lib/source-check.js (hunk 2 above)
Comment (DylanC) [reply lib/source-check.js]: I think this is kinda the same as the other comment. You want me to set isTranspiled to false if detected a longer function during the while loop?

File: lib/source-check.js (hunk 3 above)
Comment (goto-bus-stop) [reply lib/source-check.js]: yea, what I mean is that if the source map check has already set it to true, we no longer need to do the loop

Comment (DylanC) [reply lib/source-check.js]: Yeah, thats right! I'll have another look at it.

File: lib/source-check.js (hunk 2 above)
Comment (goto-bus-stop) [reply lib/source-check.js]: It wasn't intended to be the same as the other comment, but that's not a bad idea :smile:

**Comment by DylanC** [comment]
Have a look at the latest version and see what you think. 👍

File: lib/source-check.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  if (readFile.includes('//# sourceMappingURL=')) {
+    isTranspiled = true
+  } else {
+    // Loop through results and check length of fn name
+    while ((matchedObj = regex.exec(readFile)) !== null) {
+      if (matchedObj[1].length < 3) {
+        isTranspiled = true
+      } else {
+        isTranspiled = false
+      }
```
Comment (goto-bus-stop) [inline lib/source-check.js]: this if/else will now effectively treat the last function name in the source code as authoritative. flipping it to false if a long fn name is found is a good idea; then we should initialise `isTranspiled = true` right before entering the loop

Comment (DylanC) [reply lib/source-check.js]: Okay, I've removed the else and reversed the check. Hopefully this is the perfect version.

**Pull request merged**